examples/dnn_instance_segmentation_ex.cpp

```diff
@@ -0,0 +1,174 @@
+// The contents of this file are in the public domain. See LICENSE_FOR_EXAMPLE_PROGRAMS.txt
+/*
+    This example shows how to do instance segmentation on an image using net pretrained
+    on the PASCAL VOC2012 dataset.  For an introduction to what instance segmentation is,
+    see the accompanying header file dnn_instance_segmentation_ex.h.
+
+    Instructions how to run the example:
+    1. Download the PASCAL VOC2012 data, and untar it somewhere.
+       http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar
+    2. Build the dnn_instance_segmentation_train_ex example program.
+    3. Run:
+       ./dnn_instance_segmentation_train_ex /path/to/VOC2012
+    4. Wait while the network is being trained.
+    5. Build the dnn_instance_segmentation_ex example program.
+    6. Run:
+       ./dnn_instance_segmentation_ex /path/to/VOC2012-or-other-images
+
+    An alternative to steps 2-4 above is to download a pre-trained network
+    from here: http://dlib.net/files/instance_segmentation_voc2012net.dnn
+
+    It would be a good idea to become familiar with dlib's DNN tooling before reading this
+    example.  So you should read dnn_introduction_ex.cpp and dnn_introduction2_ex.cpp
+    before reading this example program.
+*/
+
+#include "dnn_instance_segmentation_ex.h"
+#include "pascal_voc_2012.h"
+
+#include <iostream>
+#include <dlib/data_io.h>
+#include <dlib/gui_widgets.h>
+
+using namespace std;
+using namespace dlib;
+ 
+// ----------------------------------------------------------------------------------------
+
+int main(int argc, char** argv) try
+{
+    if (argc != 2)
+    {
+        cout << "You call this program like this: " << endl;
+        cout << "./dnn_instance_segmentation_train_ex /path/to/images" << endl;
+        cout << endl;
+        cout << "You will also need a trained '" << instance_segmentation_net_filename << "' file." << endl;
+        cout << "You can either train it yourself (see example program" << endl;
+        cout << "dnn_instance_segmentation_train_ex), or download a" << endl;
+        cout << "copy from here: http://dlib.net/files/" << instance_segmentation_net_filename << endl;
```

Comment:
I'm uploading a sample file to Dropbox... will update the link here shortly.

Reply:
This should work: https://www.dropbox.com/s/tcimn353n3ogax7/instance_segmentation_voc2012net.dnn

Reply:
Cool.  File is live at http://dlib.net/files/instance_segmentation_voc2012net.dnn now :)